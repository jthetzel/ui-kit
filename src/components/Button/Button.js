import React from 'react';
import styled from 'react-emotion';

const Button = styled.button`
  background-color: salmon;
  border: 1px solid indianred;
  color: snow;
`;

/**
 * @component
 */
export default props => <Button {...props} />;
