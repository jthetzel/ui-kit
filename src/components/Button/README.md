Button

```js
<Button>Push Me</Button>
```