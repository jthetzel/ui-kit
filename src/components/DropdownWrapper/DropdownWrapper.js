import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import Container from './Container';

class DropdownWrapper extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isOpen: this.props.isOpen
    };

    this.toggle = this.toggle.bind(this);
    this.setChildNodeRef = this.setChildNodeRef.bind(this);
    this.handleOutsideClick = this.handleOutsideClick.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({ isOpen: nextProps.isOpen });
  }

  toggle() {
    if (!this.state.isOpen) {
      // attach/remove event handler
      document.addEventListener('click', this.handleOutsideClick, false);
    } else {
      document.removeEventListener('click', this.handleOutsideClick, false);
    }

    this.setState({ isOpen: !this.state.isOpen });
  }

  handleOutsideClick(e) {
    const isDescendantOfRoot = this.childNode && this.childNode.contains(e.target);
    if (!isDescendantOfRoot) {
      this.toggle();
    }
  }

  setChildNodeRef(ref) {
    this.childNode = ref;
  }

  render() {
    return (
      <Container innerRef={this.setChildNodeRef}>
        {this.props.opener(this.toggle, this.state.isOpen)}
        {this.state.isOpen && this.props.children}
      </Container>
    );
  }
}

DropdownWrapper.propTypes = {
  isOpen: PropTypes.bool
};

DropdownWrapper.defaultProps = {
  isOpen: false
};

export default DropdownWrapper;
