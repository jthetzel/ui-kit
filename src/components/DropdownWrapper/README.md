DropdownWrapper

```js
initialState = {
  isOpen: false
};

const onChange = () => setState({ isOpen: !state.isOpen });

<div style={{ width: 300 }}>
  <DropdownWrapper
    opener={(toggle) => (<div onClick={toggle}>{state.isOpen ? 'Close' : 'Open'}</div>)}
    onChange={onChange}
    isOpen={state.isOpen}
  >
    children
  </DropdownWrapper>
</div>
```