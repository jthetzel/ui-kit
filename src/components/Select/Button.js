import styled from 'react-emotion';

export default styled.div`
  position: relative;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
  padding: 20px 0;
  user-select: none;
  width: 100%;
  
  &:hover {
    cursor: pointer;
  }
`;
