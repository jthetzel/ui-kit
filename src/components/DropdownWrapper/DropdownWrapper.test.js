/* eslint function-paren-newline: 0 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DropdownWrapper from '../DropdownWrapper';
import Button from '../Button/Button';


test('Select does not crash', () => {
  const tree = shallow(
    <DropdownWrapper opener={value => <div>{value}</div>}>
      children
    </DropdownWrapper>
  );
  expect(toJson(tree)).toMatchSnapshot();
});

test('Select simulate opener onClick', () => {
  const wrapper = shallow(
    <DropdownWrapper
      opener={toogle => (<Button onClick={toogle}>Button</Button>)}
    >
      children
    </DropdownWrapper>
  );

  const btn = wrapper.find(Button).first();
  expect(wrapper.state().isOpen).toEqual(false);
  btn.simulate('click');
  expect(wrapper.state().isOpen).toEqual(true);
});
