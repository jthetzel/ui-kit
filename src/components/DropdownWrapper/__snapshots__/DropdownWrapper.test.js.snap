// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownWrapper does not crash 1`] = `
<Styled(div)
  innerRef={[Function]}
>
  <Styled(div)
    onClick={[Function]}
  >
    Button
  </Styled(div)>
</Styled(div)>
`;
